openupgradelib/openupgrade.py

```diff
@@ -917,6 +918,104 @@ def rename_models(cr, model_spec):
     # TODO: signal where the model occurs in references to ir_model
 
 
+def merge_models(cr, old_model, new_model, ref_field):
+    """
+    Update model references for models that have merged to an existing model.
+    :param old_model: old model
+    :param new_model: destination model
+    :param ref_field: name of the field in new model that references old model.
+
+    WARNING: This method doesn't move the records from old_model to new_model
+    tables. You should have to that previously in the migration scripts.
+    """
+    logger.info("model %s: merging to %s", old_model, new_model)
+    table = new_model.replace('.', '_')
+    renames = [
+        ('ir_attachment', 'res_model', 'res_id', ''),
+        ('ir_model_data', 'model', 'res_id', ''),
+        ('ir_filters', 'model_id', '', ''),
+        ('ir_exports', 'resource', '', ''),
+    ]
+    if is_module_installed(cr, 'mail'):
+        renames += [
+            ('mail_message', 'model', 'res_id', ''),
+            ('mail_message_subtype', 'res_model', '', ''),
+            ('mail_activity', 'res_model', 'res_id', 'res_model_id'),
+            ('mail_activity_type', '', '', 'res_model_id'),
+            ('mail_template', 'model', '', 'model_id'),
+            ('mail_alias', '', '', 'alias_model_id'),
+            ('mail_alias', '', 'alias_parent_thread_id',
+             'alias_parent_model_id'),
+            # mail_followers: special case handled below
+        ]
+    for rename in renames:
```

Comment:
I think the code would be more clear if you replaced `rename` with a tuple, perhaps something like
```
for (table, model_name_column, res_id_column, model_id_column) in renames:
```

Reply:
Ok, I like your suggestion. I will proceed with it.